examples/equilibria/coilset_opt_problem_tutorial.ex.py

```diff
@@ -0,0 +1,447 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: tags,-all
+#     notebook_metadata_filter: -jupytext.text_representation.jupytext_version
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %% tags=["remove-cell"]
+# SPDX-FileCopyrightText: 2021-present M. Coleman, J. Cook, F. Franza
+# SPDX-FileCopyrightText: 2021-present I.A. Maione, S. McIntosh
+# SPDX-FileCopyrightText: 2021-present J. Morris, D. Short
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""
+Coilset Optimisation Problem Tutorial
+"""
+
+# %% [markdown]
+# # Coilset optimisation problem tutorial
+#
+#
+# This tutorial finds a Spherical Tokamak (ST) equilibrium in a
+# double null configuration,
+# using a constrained optimisation method with bound constraints on
+# the maximum coil currents and some additional constraints.
+#
+
+# %%[markdown]
+
+# # Introduction
+
+# In this example, we will outline how to specify a `CoilsetOptimisationProblem`
+# that specifies how an 'optimised' coilset state is found during the Free Boundary
+# Equilibrium solve step.
+
+# # Imports
+
+# Import necessary equilbria module definitions.
+
+# %%
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from bluemira.equilibria.coils import Coil, CoilSet, SymmetricCircuit
+from bluemira.equilibria.equilibrium import Equilibrium
+from bluemira.equilibria.grid import Grid
+from bluemira.equilibria.optimisation.constraints import (
+    CoilFieldConstraints,
+    FieldNullConstraint,
+    IsofluxConstraint,
+    MagneticConstraintSet,
+)
+from bluemira.equilibria.optimisation.problem import (
+    TikhonovCurrentCOP,
+    UnconstrainedTikhonovCurrentGradientCOP,
+)
+from bluemira.equilibria.profiles import CustomProfile
+from bluemira.equilibria.solve import DudsonConvergence, PicardIterator
+
+# %%[markdown]
+
+# # CoilsetOptimisationProblem
+
+# The `CoilsetOptimisationProblem` class is intended to be the abstract
+# base class for coilset optimisation problems across Bluemira.
+#
+# *Remember: There are currently two optimization modules in Bluemira
+# Develop, and this example intends to illustrate the use of the newest
+# one in `bluemira.equilibria.optimisation`*
```

Comment:
We don't need to even tell people about the old one. The same reason we don't have any examples using old functionality
```suggestion
```